### DISCUSSION TITLE
post/170224-powerautomate-create-html-table/

### DISCUSSION THREAD

**Discussion by giscus** [AlexanderHenkel/blogComments#7] [Announcements]
# post/170224-powerautomate-create-html-table/

Learn how to create a dynamic table with SharePoint list data

https://alexanderhenkel.dk/post/170224-powerautomate-create-html-table/

**Comment by mjenks12** [comment]
hi, this layout is exactly what I have been looking for! I was able to incorporate 2 columns however I need several more columns, how would I add columns to this format?

**Reply by AlexanderHenkel** [reply]
**To add more columns**

First you add the rows (the data you want to display) to the **Select** action

Here I have added an **extra row**

```<td style="font-family: Open Sans, sans-serif; border: 1px solid #dededf; background-color: #ffffff; color: #000000; padding: 5px;">EXTRA ROW DATA</td>```

```
<tr>
<td style="font-family: Open Sans, sans-serif; border: 1px solid #dededf; background-color: #ffffff; color: #000000; padding: 5px;">TITLE</td>
<td style="font-family: Open Sans, sans-serif; border: 1px solid #dededf; background-color: #ffffff; color: #000000; padding: 5px;">STATUS</td>
<td style="font-family: Open Sans, sans-serif; border: 1px solid #dededf; background-color: #ffffff; color: #000000; padding: 5px;">COMPANY</td>
<td style="font-family: Open Sans, sans-serif; border: 1px solid #dededf; background-color: #ffffff; color: #000000; padding: 5px;">EXTRA ROW</td>
</tr>
```

Then you add the column headers in the **Compose headers and join rows** action

Here I added an extra column header _EXTRA ROW HEADER_

```<th style="border: 1px solid #dededf; background-color: #eceff1; color: #000000; padding: 5px;" bgcolor="#eceff1">EXTRA ROW HEADER</th>```

```
<table style="font-family: Open Sans, sans-serif; border: 1px solid #dededf; height: 100%; max-width: 50%; table-layout: fixed; border-collapse: collapse; border-spacing: 1px; text-align: left;" width="100%" height="100%" align="left">
    <tr>
    <th style="border: 1px solid #dededf; background-color: #eceff1; color: #000000; padding: 5px;" bgcolor="#eceff1">Project name</th>
    <th style="border: 1px solid #dededf; background-color: #eceff1; color: #000000; padding: 5px;" bgcolor="#eceff1">Status</th>
    <th style="border: 1px solid #dededf; background-color: #eceff1; color: #000000; padding: 5px;" bgcolor="#eceff1">Customer</th>
    <th style="border: 1px solid #dededf; background-color: #eceff1; color: #000000; padding: 5px;" bgcolor="#eceff1">EXTRA ROW HEADER</th>
    </tr>
    HERE WE ARE GOING TO INSERT AN EXPRESSION
</table>
```

**Reply by mjenks12** [reply]
Hi, thank you for this. When I copy and paste the code it works but if I try to add an additional column I am not getting the same results. I need 10 columns to replicate an accounts receivable report I would like to send in an email.

**Reply by AlexanderHenkel** [reply]
The Compose and Select action behave weird when copy pasting inside of it. I suggest you copy/paste the number of lines you need inside Notepad or Visual Studio Code, and then insert it into the actions

**Reply by mjenks12** [reply]
ok, I will try this. thank you!
